app/controllers/admin/api_controller.rb

```diff
@@ -0,0 +1,54 @@
+class Admin::ApiController < ApplicationController
+  before_filter :admin_only
+  before_filter :check_for_cancel, only: [:create, :update]
+
+  def index
```

Comment:
Oh and one more -- I would also urge that you build in a record of API accesses say in a week or month and store them with the key, and if possible make the keys sortable by # accesses -- I think we'll really want to be able to see at a glance which keys are "dead" and which ones are unexpectedly using loads -- eg if someone shares an API key or steals one and starts using it widely before we're ready to allow that, etc.